hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/TestCreateTable.scala

```diff
@@ -86,6 +87,14 @@ class TestCreateTable extends HoodieSparkSqlTestBase {
     assertResult(tableName)(tableConfig.getTableName)
     assertFalse(tableConfig.contains(OPERATION.key()))
 
+    val schemaOpt = tableConfig.getTableCreateSchema
+    assertTrue(schemaOpt.isPresent, "Table create schema should be persisted")
+    val noMetadataFields = schemaOpt.get().getFields.stream()
+      .noneMatch(new Predicate[Field] {
+        override def test(f: Field): Boolean = HoodieRecord.HOODIE_META_COLUMNS.contains(f.name())
+      })
+    assertTrue(noMetadataFields, "Table create schema should not include metadata fields")
+
```

Comment:
assertFalse(schemaOpt.get().getFields.asScala.find(f => HoodieRecord.HOODIE_META_COLUMNS.contains(f.name())).isDefined,
      "Table create schema should not include metadata fields")